package com.company;

public class Main {

    public static void main(String[] args) {
        String[][] array = new String[][]{{"5", "6", "7", "2"},
                                          {"9", "14", "6", "2"},
                                          {"7", "6", "1", "7"},
                                          {"0", "9", "3", "9"}};

        try {
            System.out.println("Сумма элементов массива " + checkArray(array));
        } catch (MyArraySizeException e) {
            e.printStackTrace();
        } catch (MyArrayDataException e) {
            e.printStackTrace();
        }
    }

    public static int checkArray(String[][] arr) throws MyArraySizeException, MyArrayDataException {
        int lengthArray = 4;
        int sum = 0;

        if (arr.length != lengthArray || arr[0].length != lengthArray)
            throw new MyArraySizeException("Размер массива " + arr.length + " x " + arr[0].length + " - это не соответствует 4x4");

        for (int i = 0; i < arr.length; i++) {
            for (int j = 0; j < arr.length; j++) {
                if (checkArrayData(arr[i][j])) {
                    sum += Integer.parseInt(arr[i][j]);
                } else {
                    throw new MyArrayDataException("В ячейке массива [" + i + "][" + j + "] лежат неверные данные - " + arr[i][j]);
                }
            }
        }
        return sum;
    }

    public static boolean checkArrayData(String a) {
        try {
            Integer.parseInt(a);
            return true;
        } catch (NumberFormatException e) {
            return false;
        }
    }
}

class MyArraySizeException extends Exception {

    public MyArraySizeException(String s) {
        System.out.println(s);
    }
}

class MyArrayDataException extends Exception {

    public MyArrayDataException(String s) {
        System.out.println(s);
    }
}
